active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,591 @@
+}
+```
+
+<a href="url"><img src="TeachingTipMarginSample.jpg" height="600"></a>
+
+### Point to a UI element
+
+You can change the teaching tip to point to a specific UI element by setting the TeachingTipPlacementMode property. For a targeted teaching tip, this placement will be relative to the UI element target. For a non-targeted teaching tip, this placement will relative to app window.  
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            Margin="15"/>
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+   AutoSaveTip.IsOpen = true;
+}
+```
+
+<a href="url"><img src="TeachingTipPointerSample.jpg" height="600"></a>
+
+### Add content
+
+Content can be added to a teaching tip using the Content property. If there is more content to show than what the size of a teaching tip will allow, a scrollbar will be automatically enabled to allow a user to scroll the content area. 
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            Margin="15">
+                <StackPanel>
+                    <CheckBox Content="Don't show tips at start up" Checked="HandleChecked" Unchecked="HandleUnchecked" />
+                    <TextBlock>You can change your tip preferences in <Hyperlink NavigateUri="app:/item/SettingsPage">Settings</Hyperlink> if you change your mind.</TextBlock>
+                </StackPanel>
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+   AutoSaveTip.IsOpen = true;
+}
+```
+
+<a href="url"><img src="TeachingTipContentSample.jpg" height="600"></a>
+
+### Add buttons
+
+A close button is automatically added to the top of a teaching tip. However, a close button with custom text that is placed at the button of a teaching tip can be used instead by setting the TeachingTipCloseButtonKind property. A custom action button can also be added by setting ActionButtonText, ActionButtonCommand and the ActionButtonCommandParameter properties. 
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            ActionButtonText="Disable"
+            ActionButtonCommand="DisableAutoSave"
+            CloseButtonKind="Footer"
```

Comment:
Is a CloseButtonKind property really necessary? What if the behavior were: if (CloseButtonText != null) show button in footer, else show the X.
Or is your intent when CloseButtonKind="Header" that something other than the X gets rendered?

Reply:
It was originally included because we were to have 4 close button kinds with the forth being "None" so that developers could implement a custom styled close button in the content area. However, now that we have decided to allow more complete styling of the close button, I believe the "None" mode can be removed and it follows to simplify the API as you have suggested. I have added this as an open question to collect any push back on this change.